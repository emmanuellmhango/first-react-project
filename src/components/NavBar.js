import React from 'react';
import { NavLink } from 'react-router-dom';

const NavBar = () => (
  <div className="header-links">
    <div className="header-title">
      <span>Math Magician</span>
    </div>
    <div className="links">
      <NavLink to="/" className="spacing link">Home</NavLink>
      {' | '}
      <NavLink to="/calculator" className="spacing link">Calculator</NavLink>
      {' | '}
      <NavLink to="/quote" className="spacing link">Quote</NavLink>
    </div>
  </div>
);

export default NavBar;
