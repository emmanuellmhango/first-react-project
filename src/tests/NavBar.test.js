import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import NavBar from '../components/NavBar';

describe('Snapshot testing', () => {
  test('Testing Home component', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
