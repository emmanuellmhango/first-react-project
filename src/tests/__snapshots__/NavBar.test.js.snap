// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing Testing Home component 1`] = `
<div
  className="header-links"
>
  <div
    className="header-title"
  >
    <span>
      Math Magician
    </span>
  </div>
  <div
    className="links"
  >
    <a
      aria-current="page"
      className="spacing link active"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
     | 
    <a
      className="spacing link"
      href="/calculator"
      onClick={[Function]}
    >
      Calculator
    </a>
     | 
    <a
      className="spacing link"
      href="/quote"
      onClick={[Function]}
    >
      Quote
    </a>
  </div>
</div>
`;
